### PR TITLE
updates to cloud migration process for better clarity

### PR DESCRIPTION
**Pull request by loujar**
- Reorganized instructions in chronological order into three main groups: 
  - steps to initialize a cloud data migration bucket
  - steps to collect snapshot data from the customer on a synchronous call
  - steps to process data migration on the cloud instance
- Added note to ensure that the latest mi2 binary is installed before proceeding with data migration (common paper cut)
- Added note to verify that all deployments/ sts have been scaled up after processing data migration to (accounting for prometheus being scaled down due to disabling alerting)
- Added note for backfilling instance to Cloud control plane after completing data migration
- various misc formatting changes

### CODE REVIEW

File: content/departments/cloud/technical-docs/v2.0/onprem_data_migration.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# On-prem data migration to Cloud v2
+# On-prem data migration to Sourcegraph Cloud
 
 > [!WARNING] This process is still a work-in-progress. For more details, see [RFC 760](https://docs.google.com/document/d/1IAgXmv2TbtU_rWXtph-KFc3qbqswREIAxO1rGALuoMQ/edit#) and the [tracking issue](https://github.com/sourcegraph/customer/issues/1525), or cc @bobheadxi (Robert Lin).
 > This page was adapted only on a best-effort basis for Cloud v2 - it has currently not yet been thoroughly tested for Cloud v2.
```
Comment (bobheadxi) [inline content/departments/cloud/technical-docs/v2.0/onprem_data_migration.md]: This page has gone through several rounds of awesome updates from you - maybe I can be removed from here now? 😄 

```suggestion
```

**Review by bobheadxi — APPROVED** [review]
This is great, thank you :)

File: content/departments/cloud/technical-docs/v2.0/onprem_data_migration.md [hunk 2]
```diff
@@ -216,18 +199,39 @@ Audit logs are generated for bucket access in the project's logs, under log entr
...
+3. Disable alerting by editing the instance `config.yaml` in [`sourcegraph/cloud`](https://github.com/sourcegraph/cloud) as follows. Make sure to deploy the terraform changes as well.
+
+```diff
+spec:
+  debug:
+-    enableAlerting: true
++    enableAlerting: false
+```
+
+4. Commit and submit your changes as a pull request. After merging and confirming the apply in Terraform Cloud, proceed with scaling down the instance:
```
Comment (michaellzc) [inline content/departments/cloud/technical-docs/v2.0/onprem_data_migration.md]: we can apply locally with `mi2 instance tfc deploy -auto-approve -force-ignore-stack-dependencies -target monitoring`

**Review by michaellzc — APPROVED** [review]

File: content/departments/cloud/technical-docs/v2.0/onprem_data_migration.md (hunk 2 above)
Comment (loujar) [reply content/departments/cloud/technical-docs/v2.0/onprem_data_migration.md]: good call out, added

**Pull request merged**